### PR TITLE
Reconcile scripts directory with brand-consolidation

### PR DESCRIPTION
**Pull request by ncksllvn**
## Description
The work for the brand-consolidation of VA.gov is currently on its own branch, `brand-consolidation`, with a modified build script for building the FE project. There was a feature flag implementation there, but it wasn't always used when it should have been, and the build itself need evolution to make it more clear how the build changes between different environments.

This PR moves all code that modifies the `options` variable in `build.js` into its own file, `scripts/options.js`. This includes -

- Function for gathering command line flags
- Function for Heroku overrides
- Function for environment overrides
- Function for brand-consolidation overrides

The goal here is that after the `options` object is established, the rest of the build can become more straightforward and we don't need to check certain build flags (primarily the brand-consolidation build flag) before each step.

The flip-side of this - where the `brand-consolidation` branch is receiving the same changes - is here, department-of-veterans-affairs/vets-website/pull/8303.

## Testing done
I've run builds on my own machine for both `npm run watch` and `npm run watch:brand-consolidation`.

## Testing Plan
Open to suggestions if there's a great way to ensure each environment builds exactly how we expect it to.

Resolves department-of-veterans-affairs/vets.gov-team/issues/12827

### CODE REVIEW

**Review by U-DON — APPROVED** [review]
LGTM. Separating the options into a new module looks much cleaner too.

**Review by U-DON — APPROVED** [review]

File: script/build.js [hunk 1]
```diff
@@ -31,113 +30,51 @@ const {
...
 
 const ignoreList = [];
-if (options.buildtype === 'production') {
+if (BUILD_OPTIONS.buildtype === 'production') {
   manifests.filter(m => !m.production).forEach(m => {
     ignoreList.push(m.contentPage);
   });
   ignoreList.push('veteran-id-card/how-to-get.md');
   ignoreList.push('veteran-id-card/how-to-upload-photo.md');
+  ignoreList.push('education/complaint-tool.md');
```
Comment (jkassemi) [inline script/build.js]: should this be added to the ignoreList in this PR?

File: script/build.js [hunk 2]
```diff
@@ -31,113 +30,51 @@ const {
...
+  ignoreList.push('disability-benefits/apply/form-526-disability-claim.md');
```
Comment (jkassemi) [inline script/build.js]: also curious for an update here if it's relevant. I was under (a potentially outdated) impression that this was meant to go live behind a password, just hidden from navigation.

File: script/collections/brand-consolidation.json [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+    }
+  },
+  "vreServiceDisabled": {
+    "pattern": "employment/vocational-rehab-and-employment/service-disabled/*md",
+    "sortBy": "order",
+    "metadata": {
+      "name": "Servicemember & Veteran Programs"
+    }
+  }
+}
```
[the remark this answers is not in the document]
Comment (jkassemi) [reply script/collections/brand-consolidation.json]: 💯

**Review by jkassemi — APPROVED** [review]
Definitely more readable. 

I'm guessing that the timestamp-based cache busting in the filename is probably the most major contributor to nondeterministic builds in this case. Should probably reevaluate that at some point, but this seems pretty straightforward.

File: script/build.js (hunk 2 above)
Comment (ncksllvn) [reply script/build.js]: Ah thank! Removing both of these from the ignore-list. Normally that list hardly ever changes, so I'm surprised there was any difference between them.

**Comment by ncksllvn** [comment]
Would someone mind also approving the brand-consolidation side of things, department-of-veterans-affairs/vets-website/pull/8306?

**Review by jbalboni — APPROVED** [review]
This looks good to me, and seems like an improvement regardless of the brand consolidation work.

**Review by wyattwalter — APPROVED** [review]
A much-needed improvement. Looks great to me!

File: script/build.js [hunk 4]
```diff
@@ -638,14 +261,14 @@ if (!options.watch && !(process.env.CHECK_BROKEN_LINKS === 'no')) {
         '/education/apply-for-education-benefits/application',
         '/pension/application/527EZ',
         '/burials-and-memorials/application/530',
-        '/health-care/apply/application',
+        '/health-care/apply-for-health-care-form-10-10ez',
```
[the remark this answers is not in the document]
Comment (ncksllvn) [reply script/build.js]: Ah we do! Good catch, appreciate it 🙂Adding it back!

**Pull request merged**